### PR TITLE
Use a new configuration approach for bootstrapping a database in SQL tutorials

### LINKED ISSUE #3672: [Config] Use a new configuration approach for bootstrapping a database in SQL tutorials
**Product:** Tarantool
**Since:** 3.0
**Root document:** see below
**SME:** @ Totktonada

# Details

Bootstrapping a database for SQL tutorials uses the old approach:

- https://www.tarantool.io/en/doc/latest/how-to/sql/sql_tutorial/#initialize
- https://www.tarantool.io/en/doc/latest/how-to/sql/improving_mysql/ (step 13)

We can use a new config instead and connect to an instance using tt.

### PR DESCRIPTION
**Pull request by xuniq**
Fixes #3672 

What has been done:

* Updated `Improving MySQL with Tarantool` page: https://github.com/tarantool/doc/pull/3891/files#diff-c243191ef89b4e031a1f3631024a625cf0e5fb513791f0ffccde63e351bc33a9L1

* Updated `SQL tutorial` page: https://github.com/tarantool/doc/pull/3891/files#diff-95e6dc7b9efa335f728ba27ba49e8b0684201521ed5c2de7747d853e72b154e4L2

* Added `mysql` directory to the `code-snippets/snippets/config` folder (used in the `Improving MySQL with Tarantool` tutorial). The directory contains `.lua` file, `config.yaml`, and `instances.yml`.
 https://github.com/tarantool/doc/pull/3891/files#diff-3fd0774b19b03018c1e7f4d5d3899e89811433354b0ab05f00d97b5f8c9a4c60R1

Staging:
- https://docs.d.tarantool.io/en/doc/gh-3672/how-to/sql/sql_tutorial/
- https://docs.d.tarantool.io/en/doc/gh-3672/how-to/sql/improving_mysql/

### CODE REVIEW

File: doc/how-to/sql/sql_tutorial.rst [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
 
-In this tutorial, the requests are done in the Tarantool interactive console.
-Start Tarantool and initialize it with :ref:`default settings <box_introspection-box_cfg>`
-by calling ``box.cfg{}``:
+After configuration, start a Tarantool instance from the tt environment directory using the following command:
 
-.. code-block:: tarantoolsession
+..  code-block:: console
+
+    $ tt connect sql_tutorial
```
Comment (xuniq) [inline doc/how-to/sql/sql_tutorial.rst]:
```suggestion
    $ tt start sql_tutorial
```

File: doc/code_snippets/snippets/mysql/instances.enabled/config.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+              listen: "3301"
+            memtx:
+              memory: 128 * 1024 * 1024
+              min_tuple_size: 16
+              max_tuple_size: 128 * 1024 * 1024
+            wal:
+              mode: "none"
+              wal_max_size: 56 * 1024 * 1024
+            snapshot:
+              by_interval: 60 * 60
```
Comment (Totktonada) [inline doc/code_snippets/snippets/mysql/instances.enabled/config.yaml]: These values are numbers, but here we have strings with expressions. It'll not be accepted.

File: doc/code_snippets/snippets/mysql/instances.enabled/config.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+              count: 6
+            log:
+              level: 7
+            fiber:
+              too_long_threshold: 0.5
```
Comment (Totktonada) [inline doc/code_snippets/snippets/mysql/instances.enabled/config.yaml]: I think that all these option are not related to the article. How about removing them?

Comment (xuniq) [reply doc/code_snippets/snippets/mysql/instances.enabled/config.yaml]: I agree with it, seems like a good suggestion. Removed these options.

File: doc/how-to/sql/improving_mysql.rst [hunk 4]
```diff
@@ -279,59 +262,66 @@ So let’s proceed.
...
+        ---
+        - - [1, 'Fluffy', 'Harold', 'cat']
+            - [2, 'Claws', 'Gwen', 'cat']
+            - [3, 'Buffy', 'Harold', 'dog']
+            - [4, 'Fang', 'Benny', 'dog']
+            - [5, 'Bowser', 'Diane', 'dog']
+            - [6, 'Chirpy', 'Gwen', 'bird']
+            - [7, 'Whistler', 'Gwen', 'bird']
+            - [8, 'Slim', 'Benny', 'snake']
+            - [9, 'Puffball', 'Diane', 'hamster']
```
Comment (Totktonada) [inline doc/how-to/sql/improving_mysql.rst]: Nit: Looks like 2-9 items are overindented.

File: doc/how-to/sql/improving_mysql.rst [hunk 5]
```diff
@@ -279,59 +262,66 @@ So let’s proceed.
...
+        $ tt connect example
 
-   .. code-block:: tarantoolsession
+#   To see the replicated data in Tarantool, run the following command:
 
-      tarantool> box.space.mysqldata:select()
+    ..  code-block:: tarantoolsession
 
-   You should see the replicated data in Tarantool!
+        tarantool> box.space.mysqldata:select()
```
Comment (Totktonada) [inline doc/how-to/sql/improving_mysql.rst]: Nit: This `#` is rendered as is (I guess it should be a number) and the code block is indented not like the previous one.

File: doc/how-to/sql/improving_mysql.rst [hunk 6]
```diff
@@ -1,275 +1,258 @@
...
-      local function bootstrap()
+        $ cd
+        $ nano /etc/tarantool/instances.available/example.lua
 
-          if not box.space.mysqldaemon then
-              s = box.schema.space.create('mysqldaemon')
-              s:create_index('primary',
-              {type = 'tree', parts = {1, 'unsigned'}, if_not_exists = true})
-          end
+#.  Replace the entire contents of the file with the following:
```
Comment (Totktonada) [inline doc/how-to/sql/improving_mysql.rst]: I guess we don't need `example.lua` anymore: there is nothing common with out resulting code. Let's just create the file w/o copying the example.

File: doc/how-to/sql/improving_mysql.rst [hunk 7]
```diff
@@ -1,275 +1,258 @@
...
-              {type = 'tree', parts = {1, 'unsigned'}, if_not_exists = true})
-          end
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/mysql/example.lua
+        :language: lua
+        :lines: 3-17
+        :dedent:
 
-      end
+    To understand more of what’s happening here, have a look at the earlier
+    `articles <https://dzone.com/articles/tarantool-101-10-steps-for-absolute-beginners-the>`_
```
Comment (Totktonada) [inline doc/how-to/sql/improving_mysql.rst]: > The article you are trying to view has been archived. Please log in to view this content.

File: doc/how-to/sql/improving_mysql.rst [hunk 8]
```diff
@@ -1,275 +1,258 @@
...
-           too_long_threshold = 0.5;
-       }
+#.  Edit the Lua example, which comes with Tarantool:
 
-      box.schema.user.grant('guest','read,write,execute','universe')
+    ..  code-block:: console
 
-      local function bootstrap()
+        $ cd
+        $ nano /etc/tarantool/instances.available/example.lua
```
Comment (Totktonada) [inline doc/how-to/sql/improving_mysql.rst]: I think that this part should be reviewed with @psergee.

AFAIU, we want to create a system wide instance and use tarantoolctl's layout here. AFAIK, `tt` supports it. However, it also may provide better ways to install things into a system.

**Review by Totktonada — APPROVED** [review]
TBH, I don't like this manual. A lot of 'put files here and there' actions are cumbersome. We should provide deb/rpm packages for such purposes and the commands should be mostly `apt-get install <...>` plus edit config plus `systemctl start <...>`.

However, I don't mind a slight update to sync it with recent tarantool versions and the tooling around. If the new steps are working, they're likely OK.

Please, review with `tt` guys and proceed.

File: doc/how-to/sql/improving_mysql.rst (hunk 8 above)
Comment (psergee) [reply doc/how-to/sql/improving_mysql.rst]: This approach does not seem to work. Yes, `tt` can work with `tarantoolctl` layout for single-script instances. But here we have a config. To make `tt` work with `config.y[a]ml` and `instances.y[a]ml`, these files must be placed in a sub-directory of `instances.enabled`:
- Instead of an `example.lua`, a user should create `/etc/tarantool/instances.available/mysql/init.lua`
- Create `config.yml` and `instances.yml` in the `mysql` dir from the step above
- Create a link in `instances.enabled` to this `mysql` directory.

File: doc/code_snippets/snippets/config/instances.enabled/mysql/config.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+groups:
+  group001:
+    replicasets:
+      replicaset001:
+        instances:
+          instance001:
+            iproto:
+              listen:
+                - uri: '127.0.0.1:3301'
```
Comment (andreyaksenov) [inline doc/code_snippets/snippets/config/instances.enabled/mysql/config.yaml]:
```suggestion
              listen:
              - uri: '127.0.0.1:3301'
```

File: doc/how-to/sql/sql_tutorial.rst [hunk 10]
```diff
@@ -30,7 +64,7 @@ Switch to the SQL language
 A feature of the client console program is that you can switch languages and
 specify the end-of-statement delimiter.
 
-Run the following commands to set the console input language to SQL and use
+Run the following commands to set the :ref:`console input <interactive_console_input_output>` language to SQL and use
```
Comment (andreyaksenov) [inline doc/how-to/sql/sql_tutorial.rst:67]: Looks like a prompt should looks smth like `sql_tutorial>`, below is the screen from the tt console for the `instance_scope` application:
image

Please take a look at other code snippets, too.

The same for MySQL, should be `mysql:instance001>` instead of `tarantool>`.

File: doc/how-to/sql/improving_mysql.rst [hunk 11]
```diff
@@ -1,275 +1,250 @@
...
+        mysql -u root -p menagerie < load_pet_tbl.sql
+        mysql menagerie -u root -p < ins_puff_rec.sql
+        mysql menagerie -u root -p < cr_event_tbl.sql
 
-#. Let’s enter the MySQL console now and massage the data for use with the
-   Tarantool replicator (we are adding an ID, changing a field name to avoid
-   conflict, and cutting down the number of fields; note that with real data,
-   this is the step that will involve the most tweaking):
+#.  Enter MySQL console and massage the data for use with the
+    Tarantool replicator. On this step, you:
```
Comment (andreyaksenov) [inline doc/how-to/sql/improving_mysql.rst]:
```suggestion
    Tarantool replicator. In this step, you:
```

File: doc/how-to/sql/improving_mysql.rst [hunk 12]
```diff
@@ -1,275 +1,250 @@
...
-   At the bottom you will see "mysqldaemon" and "mysqldata" spaces. Then exit with "CTRL+C".
+    ..  code-block:: tarantoolsession
 
-#. Now that we have MySQL and Tarantool set up, we can proceed to configure
-   our replicator. First let’s work with ``replicatord.yml`` in the main
-   ``tarantool-mysql-replication`` directory.
+        tarantool> box.space._space:select()
 
-   .. code-block:: bash
+    At the bottom you will see ``mysqldaemon`` and ``mysqldata`` spaces. Then exit with "CTRL+C".
```
Comment (andreyaksenov) [inline doc/how-to/sql/improving_mysql.rst]:
```suggestion
    At the bottom, you will see ``mysqldaemon`` and ``mysqldata`` spaces. Then exit with "CTRL+C".
```

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

File: doc/code_snippets/snippets/config/instances.enabled/mysql/config.yaml (hunk 9 above)
Comment (xuniq) [reply doc/code_snippets/snippets/config/instances.enabled/mysql/config.yaml]: Fixed

File: doc/how-to/sql/sql_tutorial.rst (hunk 10 above)
Comment (xuniq) [reply doc/how-to/sql/sql_tutorial.rst:67]: Thanks! Fixed.

**Review by andreyaksenov — APPROVED** [review]
